main/src/main/java/net/citizensnpcs/EventListen.java

```diff
@@ -481,6 +481,18 @@ public void onPlayerInteractEntity(PlayerInteractEntityEvent event) {
     @EventHandler(priority = EventPriority.MONITOR)
     public void onPlayerJoin(PlayerJoinEvent event) {
         skinUpdateTracker.updatePlayer(event.getPlayer(), 6 * 20, true);
+
+        for (NPC npc : getAllNPCs()) {
+            String teamName = npc.data().get(NPC.SCOREBOARD_FAKE_TEAM_NAME_METADATA, "");
+            Team team = null;
+            if (!(npc.getEntity() instanceof Player) || teamName.length() == 0

```

Comment:
Reorder the if statement. Also to be honest now the PlayerUpdateTask tracks all spawned player NPCs perhaps it's better to make a global NPC player list?

Reply:
Reorder how? For reference this `if` line is exactly copypasted from `CitizensNPC#updateCustomName`

Reply:
Yeah but it's being run in a loop now so put the teamName check first